bootstrapWorktree: generate dist/parse5.mjs (+ minimal build artifacts) so worktree can run full unit suite

## Context

`ai/scripts/bootstrapWorktree.mjs` currently scopes to: gitignored `ai/mcp/server/<name>/config.mjs` copies + gitignored `.neo-ai-data/` data subdirs symlinks. It does NOT generate build artifacts in the worktree's `dist/` directory.

**Empirical anchor 2026-05-10:** running `CI=true npm run test-unit` (full suite) in a Claude Code worktree fails with:
```
Error: Cannot find module '/path/to/.claude/worktrees/<name>/dist/parse5.mjs'
  imported from /path/to/.claude/worktrees/<name>/src/functional/util/HtmlTemplateProcessor.mjs
```

Operator-flagged hypothesis: *"dist/parse5 probably get generated by build all. might be a `ai/scripts/bootstrapWorktree.mjs` item. mostly important for the left hemisphere, but could always be done (takes less than 30s on this machine)."*

Verified:
- `dist/parse5.mjs` generated by `npm run bundle-parse5` per `buildScripts/README.md`
- `buildScripts/util/templateBuildProcessor.mjs:2` imports `from '../../dist/parse5.mjs'`
- Main checkout `/Users/Shared/github/neomjs/neo/dist/` has the full set: parse5, esm/, development/, production/, highlight/
- Fresh worktree `dist/` has only `ai-knowledge-base.jsonl` (bootstrapped per current scope)
- Workaround today: scope test runs to AI-only substrate (`npm run test-unit -- test/playwright/unit/ai/`) which avoids parse5-dependent specs

## The Problem

Without dist artifacts in a fresh worktree:
1. Full unit-suite verification (`CI=true npm run test-unit`) fails immediately
2. Any spec importing `HtmlTemplateProcessor` (functional component substrate) fails
3. Workaround: narrower spec-pattern scoping — but this leaves the worktree without true CI-equivalent verification capability
4. Recurring friction across Claude Code worktree sessions; operator-flagged today after running into it for the second time this session

## The Architectural Reality

`bootstrapWorktree.mjs` distinguishes (per its own JSDoc):
- **Source code:** must be REAL COPIES (symlinks cause `Neo.setupClass` namespace collisions when worktree-local module + main-checkout module both register)
- **Data directories:** SAFE TO SYMLINK (no ESM import chains; better-sqlite3 + Chroma open by path transparently)

`dist/parse5.mjs` is a special case — it's a BUNDLED THIRD-PARTY module (not a Neo class), but it IS ESM-imported by `templateBuildProcessor.mjs`. Per the rule's spirit (namespace-collision), parse5 itself doesn't trigger collisions (it doesn't extend Neo.core.Base or register namespaces), so symlinking IS technically safe. But **generating dist artifacts in the worktree** is cleaner — keeps worktree self-contained, removes semantic-edge-case risk.

Same applies to other dist artifacts (esm/, development/, production/, highlight/) — bundled outputs from various `npm run build-*` scripts, ESM-imported by tests + runtime.

## The Fix

Extend `ai/scripts/bootstrapWorktree.mjs` with a dist-generation step:

```js
// After config copy + data symlinks + gitignored-file symlinks:
console.log('[bootstrap] Generating dist artifacts...');
spawnSync('npm', ['run', 'bundle-parse5'], {cwd: workTreeRoot, stdio: 'inherit'});
// Optionally: trigger broader build-all if operator wants full dist parity with main checkout
```

**Scope decision points:**
1. **Minimal scope:** just `bundle-parse5` (sufficient for unit-suite-runs that import HtmlTemplateProcessor)
2. **Full scope:** entire `build-all` run (covers all dist artifacts; takes ~few minutes; matches main-checkout parity exactly)
3. **Opt-in via CLI flag:** `--build-dist` flag, off by default (minimizes default bootstrap time)

Operator-framing favors option 1 (<30s) or option 3 (opt-in for full).

## Acceptance Criteria

- [ ] **AC1:** `bootstrapWorktree.mjs` generates `dist/parse5.mjs` (at minimum) post-config-copy + post-data-symlinks
- [ ] **AC2:** Fresh worktree post-bootstrap can run `CI=true npm run test-unit` without "Cannot find module dist/parse5.mjs" error (verify empirically)
- [ ] **AC3:** Existing bootstrap behavior unchanged (config copy + data symlinks + gitignored-file symlinks all preserved)
- [ ] **AC4 (scope decision-AC):** Either default-on minimal-scope (just bundle-parse5) OR opt-in CLI flag for broader build-all — implementation-time decision
- [ ] **AC5:** JSDoc updated to document the new dist-generation step + scope decision rationale

## Out of Scope

- **Symlinking dist/ to main checkout** — rejected; mixes data semantics with build artifacts; potential edge cases with namespace collisions for any ESM-imported dist module
- **Full build-all by default** — too aggressive; minimal scope is sufficient for unblocking unit-suite + keeps default bootstrap time bounded
- **Generating per-app build outputs** — too aggressive; only platform-bundled-modules (parse5) needed for test substrate

## Avoided Traps

- **Skip-as-low-priority** — operator-flagged today after empirical re-occurrence; small scope; high recurring-friction-reduction ROI for every Claude Code worktree session
- **Bundle into broader bootstrap refactor** — minimal change; bounded scope; no need to expand
- **Symlink instead of generate** — symlink semantics for ESM-imported modules has namespace-collision risk per bootstrap's own JSDoc rule, even though parse5 specifically wouldn't trigger it. Generation is cleaner architecturally.

## Related

- **Operator hypothesis surface:** 2026-05-10 session, *"dist/parse5 probably get generated by build all. might be a ai/scripts/bootstrapWorktree.mjs item."*
- **Empirical recurrence:** 2 cases this session of full-unit-suite invocations failing on dist/parse5.mjs missing
- **bootstrap JSDoc source-vs-data symlink rule:** `ai/scripts/bootstrapWorktree.mjs` head comment
- **parse5 build script:** `buildScripts/README.md` + `npm run bundle-parse5`

Origin Session ID: c2912891-b459-4a03-b2af-154d5e264df1

Retrieval Hint: `query_raw_memories(query="bootstrapWorktree dist parse5 generation unit suite full-run worktree friction")`

--- a/ai/scripts/bootstrapWorktree.mjs
+++ b/ai/scripts/bootstrapWorktree.mjs
@@ -461,15 +461,11 @@ export async function symlinkGitignoredFiles({
  *
  * Worktrees off `origin/dev` start without `node_modules` (gitignored) AND without
  * `dist/parse5.mjs` (gitignored test-runner prerequisite). Both are needed to run the
- * Playwright unit-test suite or any SDK-consuming script. Adding these as opt-in flags
- * here closes the manual-multi-step bootstrap gap surfaced empirically during #10339
- * implementation (per #10351).
+ * Playwright unit-test suite or any SDK-consuming script.
  *
  * Idempotent: skips `npm install` when `node_modules/` is already present (e.g.,
- * symlinked from main, prior `--install` invocation, or manual `npm i`). Always runs
- * `npm run bundle-parse5` because the bundle output lives under `dist/` (gitignored)
- * and is cheap to rebuild — surfacing it explicitly under `--install` closes the
- * "test runner fails with `Cannot find module '.../dist/parse5.mjs'`" friction.
+ * symlinked from main, or manual `npm i`). Always runs `npm run bundle-parse5` 
+ * because the bundle output lives under `dist/` (gitignored) and is cheap to rebuild.
  *
  * Cost anchor: ~17s for `npm install` on a populated local cache (808 packages,
  * empirically observed during #10339 implementation, 2026-04-26). `bundle-parse5`
@@ -507,8 +503,12 @@ export async function installDependencies({projectRoot, log = console.log, exec 
 /**
  * @summary Runs the full `npm run build-all` after ensuring dependencies are installed.
  *
- * Implies {@link installDependencies}. Required for tickets that touch the frontend
- * Webpack distributions or themes — backend-only / MCP-only tickets do not need this.
+ * Implies {@link installDependencies}. Default behavior for fresh worktrees (per #11163).
+ *
+ * **Scope Decision (#11163):** Rather than generating only `parse5` minimally, the operator 
+ * preferred a full `build-all` invocation as the default bootstrap step to ensure *all* 
+ * distributions (ESM, themes, workers, highlight, parse5) are ready. It resolves "Cannot find 
+ * module dist/parse5.mjs" friction for test suites in <30s on M-series hardware.
  *
  * NOT idempotent in the same idempotent-skip sense as the other bootstrap helpers —
  * `npm run build-all` re-runs every invocation. Webpack itself caches incremental builds,

